sql/descriptor.go

```diff
@@ -72,6 +72,32 @@ func (p *planner) writeDescriptor(key proto.Key, descriptor descriptorProto, ifN
 	})
 }
 
+// deleteDescriptor delete the descriptor at `key`.
+func (p *planner) deleteDescriptor(key proto.Key) error {
+	// TODO(pmattis): The error currently returned below is likely going to be
+	// difficult to interpret.
+	// TODO(XisiHuang): Need to handle if-exists here as well.
+	return p.db.Txn(func(txn *client.Txn) error {
+		// Check whether key exists.
+		gr, err := txn.Get(key)
+		if err != nil {
+			return err
+		}
+
+		if !gr.Exists() {
```

Comment:
There is overlap here with `planner.DropTable` which has already retrieved the descriptor bytes. Perhaps this method isn't necessary and you can just inline the deletions below into `DropTable`.